src/VisualStudio/RemoteHostClientMock/Remote/RemoteHostClientFactory.cs

```diff
@@ -21,7 +21,7 @@ internal class RemoteHostClientFactory : IRemoteHostClientFactory
                 // this is the point where we can create different kind of remote host client in future (cloud or etc)
                 if (workspace.Options.GetOption(RemoteHostClientFactoryOptions.RemoteHost_InProc))
                 {
-                    return await InProcRemoteHostClient.CreateAsync(workspace, cancellationToken).ConfigureAwait(false);
+                    return await InProcRemoteHostClient.CreateAsync(workspace, forTesting: false, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
This seems like the only place where we are using `forTesting = false`. Couple of suggestions:

1. Can we move the RemoteHostClientMock under some test folder? Currently the folder structure `src/VisualStudio/RemoteHostClientMock/` makes it seem like it is shipping product code.
2. Can you add comments on why we are setting this flag to false here?